site/src/components/navbar/mobile-nav-items.tsx

```diff
@@ -185,6 +170,21 @@ const MobileNavNestedPage = <T extends SiteMapPage | SiteMapPageSection>({
                 />
               ),
             )}
+            {itemIsPage(item) ? (
```

Comment:
Not sure the order was incorrect previously, but it was after the changes introduced by this PR.

The sub-sections of a page should be displayed before its sub-pages, so that for the "Specification" page the "Introduction" sub-section gets displayed before all the sub-pages:
<img width="504" alt="image" src="https://user-images.githubusercontent.com/42802102/177155022-1e009f28-b4f2-4bb9-ba7c-0c4fafd08935.png">